File: app/src/main/java/chat/rocket/android/layouthelper/chatroom/ModelViewHolder.java
package chat.rocket.android.layouthelper.chatroom;

import android.support.v7.widget.RecyclerView;
import android.view.View;

public abstract class ModelViewHolder<T> extends RecyclerView.ViewHolder {

  public ModelViewHolder(View itemView) {
    super(itemView);
  }

  public abstract void bind(T model);
}

File: app/src/main/java/chat/rocket/android/layouthelper/chatroom/ModelViewHolder.kt
package chat.rocket.android.layouthelper.chatroom

import android.support.v7.widget.RecyclerView
import android.view.View

/**
 * ViewHolder for Java models.
 */
abstract class ModelViewHolder<in T>(itemView: View) : RecyclerView.ViewHolder(itemView) {

    /**
     * Binds the model.
     */
    abstract fun bind(model: T, autoLoadImage: Boolean)
}